sys/log/full/include/log/log.h

```diff
@@ -79,12 +79,12 @@ typedef int (*lh_read_mbuf_func_t)(struct log *, void *dptr, struct os_mbuf *om,
                                    uint16_t offset, uint16_t len);
 typedef int (*lh_append_func_t)(struct log *, void *buf, int len);
 typedef int (*lh_append_body_func_t)(struct log *log,
-                                     const struct log_entry_hdr *hdr,
+                                     struct log_entry_hdr *hdr,
                                      const void *body, int body_len);
-typedef int (*lh_append_mbuf_func_t)(struct log *, const struct os_mbuf *om);
+typedef int (*lh_append_mbuf_func_t)(struct log *, struct os_mbuf *om);
 typedef int (*lh_append_mbuf_body_func_t)(struct log *log,
-                                          const struct log_entry_hdr *hdr,
-                                          const struct os_mbuf *om);
+                                          struct log_entry_hdr *hdr,
+                                          struct os_mbuf *om);
```

Comment:
Is it necessary to remove `const` from all the header parameters in the log API?  I understand why it had to be removed from the mbuf parameters (so that the log functions can call `os_mbuf_pullup()`), but I think the headers could remain const.  I would be reluctant to change the log API more than is necessary.

Reply:
Agreed, changed, I was reading the image hash in the respective append functions and hence, the log entry header could not be made const. Since that has changed and only get's called from `append_prepare()`, I can just change it back to be const.